cli/.js/src/main/scala/aqua/ArgOpts.scala

```diff
@@ -96,18 +119,31 @@ object ArgOpts {
       case None if vars.nonEmpty =>
         invalidNec("Function have non-literals, so, data should be presented")
       case None =>
-        validNec(Map.empty)
+        validNec((cliFunc, Map.empty))
       case Some(data) =>
-        val services = vars.map { vm =>
+        vars.map { vm =>
           val arg = {
             val a = data.selectDynamic(vm.name)
             if (js.isUndefined(a)) null
             else a
           }
 
-          vm.name -> ArgumentGetter(vm, arg)
-        }
-        validNec(services.toMap)
+          val typeV = jsTypeToAqua(vm.name, arg)
+
+          typeV.map(t => (vm.copy(baseType = t), arg))
+        }.sequence
+          .map(_.map { case (vm, arg) =>
+            vm.name -> ArgumentGetter(vm, arg)
+          }.toMap)
+          .andThen { services =>
+            val argsWithTypes = cliFunc.args.map {
+              case v @ VarRaw(n, t) =>
+                services.get(n).map(g => v.copy(baseType = g.function.value.baseType)).getOrElse(v)
```

Comment:
I cannot understand this line

Reply:
If there is a service with such name, get type from this service and put it to argument

Reply:
Why? Can you add a comment to this line?